src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2338,21 +2376,41 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     //   have a visible character.
     winrt::Windows::Foundation::Size TermControl::MinimumSize()
     {
-        const auto fontSize = _actualFont.GetSize();
-        double width = fontSize.X;
-        double height = fontSize.Y;
-        // Reserve additional space if scrollbar is intended to be visible
-        if (_settings.ScrollState() == ScrollbarState::Visible)
+        if (_initializedTerminal)
         {
-            width += ScrollBar().ActualWidth();
-        }
+            const auto fontSize = _actualFont.GetSize();
+            double width = fontSize.X;
+            double height = fontSize.Y;
+            // Reserve additional space if scrollbar is intended to be visible
+            if (_settings.ScrollState() == ScrollbarState::Visible)
+            {
+                width += ScrollBar().ActualWidth();
+            }
 
-        // Account for the size of any padding
-        const auto padding = GetPadding();
-        width += padding.Left + padding.Right;
-        height += padding.Top + padding.Bottom;
+            // Account for the size of any padding
+            const auto padding = GetPadding();
+            width += padding.Left + padding.Right;
+            height += padding.Top + padding.Bottom;
 
-        return { gsl::narrow_cast<float>(width), gsl::narrow_cast<float>(height) };
+            return { gsl::narrow_cast<float>(width), gsl::narrow_cast<float>(height) };
+        }
+        else
+        {
+            // If the terminal hasn't been initialized yet, then the font size will
+            // have dimensions {1, fontSize.Y}, which can mess with consumers of
+            // this method. In that case, we'll need to pre-calculate the font
+            // width, before we actually have a renderer or swapchain.
+            const winrt::Windows::Foundation::Size minSize{ 1, 1 };
+            const double scaleFactor = DisplayInformation::GetForCurrentView().RawPixelsPerViewPixel();
+            const auto dpi = ::base::saturated_cast<uint32_t>(USER_DEFAULT_SCREEN_DPI * scaleFactor);
+            return GetProposedDimensions(minSize,

```

Comment:
Just to make sure I understand this correctly - this will end up giving us something closer to the dimensions of {fontSize.X, fontSize.Y} (plus padding, scrollbar) since the size given is {1,1}? And so now that it isn't always so skewed towards fontSize.Y, panes will be able to split more consistently?

Reply:
Yep, that's basically it. Before this change in TermControl, you could use a commandline to keep splitting panes far past the actual minimum size, because we'd still have something like 0 for `fontSize.X`.